src/ol/featureloader.js

```diff
@@ -87,8 +87,10 @@ ol.featureloader.tile = function(url, format) {
        * @this {ol.VectorTile}
        */
       function(features, dataProjection) {
-        this.setProjection(dataProjection);
-        this.setFeatures(features);
+        /** @type {ol.VectorTile} */
+        var me = this;
```

Comment:
I don't see any reason why `ol.featureloader.tile` needs to be in this module.  This could be moved to the only place where it is used (in `src/ol/source/vectortile.js`).  I'd make it a method of `ol.source.VectorTile`.

I also don't see any place where `ol.featureloader.loadFeaturesXhr` is used.  Looks like this whole file could be removed.

Please don't add workarounds for behavior we don't understand about the compiler.